### ISSUE TITLE
BUG. when I use checkout I get the error "worktree contains unstaged changes"

### ISSUE THREAD

**Issue by LazarenkoA** [go-git/go-git#511]
code
```go
wt, _ := rep.Worktree()
err := wt.Checkout(&git.CheckoutOptions{
		Hash:   plumbing.Hash{},
		Branch: plumbing.NewBranchReferenceName(br),
})
```

I get the error "worktree contains unstaged changes" but the branch is changed (can see it in git bash). And for the record, the error doesn't appear if I change the brunch by the command in git bash

**Comment by LazarenkoA** [comment]
`wt.Status()` return a list of changes, but why the command `git status` in git bash return `nothing to commit, working tree clean`?

**Comment by LazarenkoA** [comment]
with the flag `Force: true` checkout occurs, but for a very long time, in general, I noticed that the log is also taking a long time

**Comment by maragunde93** [comment]
I am having the same issue when using Kaniko project (which relies on this library). I still couldn't find why is this happening only in 1 of our repositories, in fact it is happening only in 1 branch.

**Comment by maragunde93** [comment]
I found the solution for this. There was a symlink in the repository (pushed to the repo) which was causing the checkout to get bugged. I think this is only a work around.

For anyone with the same issue, delete the symlink running this command on the root of the project: `find -type l -delete`. Then commit and push the changes. 

Even though I made this change on the last commit, it fixed the checkout for any commit in the branch.
